linera-chain/src/chain.rs

```diff
@@ -650,28 +664,29 @@ where
                     block_timestamp: timestamp,
                 }
             );
-            let bundles = bundles_by_origin.entry(origin).or_default();
+            let bundles = bundles_by_origin.entry(*origin).or_default();
             bundles.push(bundle);
         }
-        let origins = bundles_by_origin.keys().copied();
-        let inboxes = self.inboxes.try_load_entries_mut(origins).await?;
+        let origins = bundles_by_origin.keys().copied().collect::<Vec<_>>();
+        let inboxes = self.inboxes.try_load_entries_mut(&origins).await?;
         let mut removed_unskippable = HashSet::new();
         for ((origin, bundles), mut inbox) in bundles_by_origin.into_iter().zip(inboxes) {
             tracing::trace!(
-                "Removing {:?} from {chain_id:.8}'s inbox for {origin:}",
+                "Removing [{}] from {chain_id:.8}'s inbox for {origin:}",
```

Comment:
I personally prefer if arguments of the log (things that can change) are not inlined to the log line:
```rust
tracing::trace!(
  bundle_heights: bundles.iter()...,
  origin_chain = ?origin,
  "Removing bundles from chain's inbox ....
)
```

not something that has to be done in this PR ofc.